### ISSUE TITLE
Creating unsynced pattern from pattern editor results in empty pattern

### ISSUE THREAD

**Issue by nosilver4u** [WordPress/gutenberg#53319]
### Description

If one wants to create an unsynced pattern from an existing reusable block (now a synced pattern), there seem to be a couple (cumbersome) ways to do so. One of the ways we tried is to edit a synced pattern from the pattern manager, and then create a new unsynced pattern from the top-level group. This successfully creates a new unsynced pattern, but the new pattern has no content at all.

### Step-by-step reproduction instructions

1. Create a re-usable block. In my test, I created one that was a group block containing a heading and an image block.
2. From the re-usable block, choose Manage patterns.
3. This takes you to the pattern editor, in the List View, click the dotted menu icon for the top-level group block, and create a new unsynced block.
4. Go back to the pattern manager and edit the new block, there is no content. Likewise, one could insert the new unsynced block in a post to see that no content is added.

### Environment info

WP 6.3 RC3, Chrome (latest), Mac OS X 13.4.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by stacimc** [comment]
I was able to reproduce this according to the instructions. The newly created pattern was empty regardless of whether I tried to create a synced or unsynced pattern from a reusable block:

https://github.com/WordPress/gutenberg/assets/63313398/5d4f7d74-286a-4edf-97bb-daf13954e4da

This only happened for me if the source pattern was a synced pattern/reusable block. If the pattern was unsynced, I was able to create new synced and unsynced patterns from it without the inner blocks disappearing.

**Comment by glendaviesnz** [comment]
I was able to replicate this. It should not be possible to create a pattern from a synced pattern. In the post editor the Synced pattern will only show the `manage patterns` or `detach patterns` option, but for some reason in some circumstances the the create pattern menu is shown in the list view:

https://github.com/WordPress/gutenberg/assets/3629020/1027aa4d-69bc-4360-88df-9a1494c76102

**Comment by nosilver4u** [comment]
@glendaviesnz Do note, that the issue I reported is specifically about what happens when one is editing a pattern after having clicked "Manage patterns". What you showed in your video is very odd though, and I think should also be addressed.

I could see the potential for Create pattern in that menu being used to create an unsynced pattern as a clone of the original synced pattern (related to something I requested in a separate issue). But in this context, I think it would cause more confusion than anything else, and Create pattern should not be displayed in the menu for a synced pattern.
